src/backends/torch/torchlib.cc

```diff
@@ -610,7 +631,16 @@ namespace dd
                       throw MLLibBadParamException("unknown loss " + _loss);
                   }
               }
-            else
+            else if (_regression)
+              {
+                if (_loss.empty() || _loss == "L1" || _loss == "l1")
+                  loss = torch::l1_loss(y_pred, y);
+                else if (_loss == "L2" || _loss == "l2" || _loss == "eucl")
+                  loss = torch::mse_loss(y_pred, y);
+                else
+                  throw MLLibBadParamException("unknown loss " + _loss);
+              }
+            else // classification
```

Comment:
It's often better to be explicit about this, like:
```
else if (_classification){
    ...
} else {
     throw MLLibBadParamException("unexpected model type");
}
```

Reply:
done